packages/core/src/YarnTool.ts

```diff
@@ -0,0 +1,93 @@
+import path from 'path';
+import fs from 'fs-extra';
+
+import { Tool, ToolType, Package, PackageJSON, Packages, InvalidMonorepoError } from './Tool';
+import { expandPackageGlobs, expandPackageGlobsSync } from "./expandPackageGlobs";
+
+export interface YarnPackageJSON extends PackageJSON {
+    workspaces?: string[] | { packages: string[] };
+}
+
+export const YarnTool : Tool = {
+    type: 'yarn',
+
+    async isMonorepoRoot(directory: string): Promise<boolean> {
+        try {
+            const pkgJson = (await fs.readJson(path.join(directory, "package.json"))) as YarnPackageJSON;
+            if (pkgJson.workspaces) {
+                if (Array.isArray(pkgJson.workspaces) || Array.isArray(pkgJson.workspaces.packages)) {
+                    return true;
+                }
+            }
+        } catch (err) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+        return false;
+    },
+
+    isMonorepoRootSync(directory: string): boolean {
+        try {
+            const pkgJson = fs.readJsonSync(path.join(directory, "package.json")) as YarnPackageJSON;
+            if (pkgJson.workspaces) {
+                if (Array.isArray(pkgJson.workspaces) || Array.isArray(pkgJson.workspaces.packages)) {
+                    return true;
+                }
+            }
+        } catch (err) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+        return false;
+    },
+
+    async getPackages(directory: string): Promise<Packages> {
+        const rootDir = path.resolve(directory);
+
+        try {
+            const pkgJson = (await fs.readJson(path.join(directory, "package.json"))) as YarnPackageJSON;
+            const packageGlobs: string[] = Array.isArray(pkgJson.workspaces) ? pkgJson.workspaces : pkgJson.workspaces!.packages;
```

Comment:
Similar doubt here as with Bolt before, this could make this function throw a `TypeError` instead of `InvalidMonorepoError`. Should we use the same strategy as with Bolt?
https://github.com/Thinkmill/manypkg/blob/3191db736c686119f3c3da9971872792bf7b8775/packages/core/src/BoltTool.ts#L49-L51